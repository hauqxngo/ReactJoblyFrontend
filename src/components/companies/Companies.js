import React from "react";
import { Link } from "react-router-dom";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";

/** Show some info of a company on the Company List
 *
 * CompanyList -> Companies
 */

const Companies = ({ name, description, handle }) => {
  return (
    <div className="CompanyCard">
      <Link className="text-decoration-none" to={`/companies/${handle}`}>
        <Card body color="secondary mx-5 my-2" outline>
          <CardBody>
            <CardTitle className="text-success" tag="h5">
              {name}
            </CardTitle>
            <CardText className="text-dark">{description}</CardText>
          </CardBody>
        </Card>
      </Link>
    </div>
  );
};

export default Companies;
