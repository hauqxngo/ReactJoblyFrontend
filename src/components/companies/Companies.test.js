import React from "react";
import { render } from "@testing-library/react";
import Companies from "./Companies";
import { MemoryRouter } from "react-router";

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Companies
        handle="abc"
        name="ABC Company"
        description="A is for Apple. B is for Bus. C is for Cat."
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
