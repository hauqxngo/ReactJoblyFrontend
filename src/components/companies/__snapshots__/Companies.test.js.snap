// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard"
  >
    <a
      class="text-decoration-none"
      href="/companies/abc"
    >
      <div
        class="card card-body border-secondary mx-5 my-2"
      >
        <div
          class="card-body"
        >
          <h5
            class="text-success card-title"
          >
            ABC Company
          </h5>
          <p
            class="text-dark card-text"
          >
            A is for Apple. B is for Bus. C is for Cat.
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
